### PR TITLE
fix: fetch workflows on the backend

### PR DESCRIPTION
**Pull request by Udit-takkar**
## What does this PR do?

- Fixes #XXXX (GitHub issue number)
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)
Fixes https://github.com/calcom/cal.com/pull/15083/files#r1648373166

Screenshot 2024-07-16 at 5 43 09 PM

Reported here:- https://app.campsite.co/cal/posts/025cr6o9p6er

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com). If N/A, write N/A here and check the checkbox.
- [x] N/A I confirm automated tests are in place that prove my fix is effective or that my feature works.

### CODE REVIEW

File: apps/web/modules/event-types/views/event-types-single-view.tsx [hunk 1]
```diff
@@ -859,7 +859,7 @@ const EventTypePageWrapper: React.FC<PageProps> & {
 
   const { data: workflows } = trpc.viewer.workflows.getAllActiveWorkflows.useQuery({
     eventType: {
-      workflows: eventType.workflows,
```
Comment (Udit-takkar) [inline apps/web/modules/event-types/views/event-types-single-view.tsx:862]: Do not pass workflows

File: packages/trpc/server/routers/viewer/workflows/util.ts [hunk 2]
```diff
@@ -804,3 +807,85 @@ export async function getAllWorkflowsFromEventType(
...
+            {
+              userId: userId,
+            },
+          ],
+        },
+        {
+          id: eventTypeId,
+        },
+      ],
+    },
```
Comment (Udit-takkar) [inline packages/trpc/server/routers/viewer/workflows/util.ts:845]: same logic as getEventTypeById. TODO: move this to repository pattern

Comment (keithwillcode) [reply packages/trpc/server/routers/viewer/workflows/util.ts:845]: Why not put it in a repository now?

Comment (Udit-takkar) [reply packages/trpc/server/routers/viewer/workflows/util.ts:845]: the return type inside `getEventTypeById` is a bit complex and require too many relational fields data which is not required here and might slow down the result and we'll have to use Prisma.validator for all nested fields to make sure type checks pass

Comment (Udit-takkar) [reply packages/trpc/server/routers/viewer/workflows/util.ts:845]: let me try adding this to repository.

File: packages/lib/event-types/getEventTypeById.ts [hunk 3]
```diff
@@ -46,216 +47,7 @@ export const getEventTypeById = async ({
...
-              },
-              steps: true,
-            },
-          },
-        },
-      },
-      secondaryEmailId: true,
-    },
-  });
+  const rawEventType = await EventTypeRepository.findById({ id: eventTypeId, userId });
```
Comment (joeauyeung) [inline packages/lib/event-types/getEventTypeById.ts:50]: Love the refactor here.

**Review by joeauyeung — APPROVED** [review]
Tested with both individual, team, and org workflows. Great work @Udit-takkar

File: packages/lib/server/repository/eventType.ts [hunk 4]
```diff
@@ -194,6 +194,231 @@ export class EventTypeRepository {
...
+          id: true,
+          subscriberUrl: true,
+          payloadTemplate: true,
+          active: true,
+          eventTriggers: true,
+          secret: true,
+          eventTypeId: true,
+        },
+      },
+      workflows: {
```
[the remark this answers is not in the document]
Comment (Udit-takkar) [reply packages/lib/server/repository/eventType.ts:344]: `getEventTypeById` is also being used in API v2 so i don't we should change the data now.

**Pull request merged**